staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1610,6 +1643,25 @@ func extractNodeAddresses(instance *ec2.Instance) ([]v1.NodeAddress, error) {
 		}
 	}
 
+	// IPv6. Ordered after IPv4 addresses, so legacy code can continue to just use the "first" address.
+	for _, networkInterface := range instance.NetworkInterfaces {
+		// skip network interfaces that are not currently in use
+		if aws.StringValue(networkInterface.Status) != ec2.NetworkInterfaceStatusInUse {
+			continue
+		}
+
+		for _, addr6 := range networkInterface.Ipv6Addresses {
+			if ipAddress := aws.StringValue(addr6.Ipv6Address); ipAddress != "" {
+				ip := netutils.ParseIPSloppy(ipAddress)
+				if ip == nil {
+					return nil, fmt.Errorf("EC2 instance had invalid IPv6 address: %s (%q)", aws.StringValue(instance.InstanceId), ipAddress)
+				}
+				// NB: "Internal" is actually about intra-cluster reachability, and not public vs private.
```

Comment:
is aws returning link local addresses, do you want to filter here to obtain only reachable addresses?